backend/src/payments/services/payments.service.ts

```diff
@@ -23,16 +23,22 @@ export class PaymentsService {
     private readonly userService: UserDBService,
   ) {}
 
-  async createCheckoutSession(plan: string, userId: Schema.Types.ObjectId) {
+  async createCheckoutSession(
+    plan: string,
+    userId: Schema.Types.ObjectId,
+  ): Promise<Stripe.Response<Stripe.Checkout.Session>> {
     const user = await this.userService.findOneById(userId)
     if (!user)
       throw new UnauthorizedException(
         'This user does not exist and cannot make a purchase', // How unfortunate, we always want money
       )
 
-    if (plan === user.paymentPlan)
+    if (plan === user.paymentPlan) {
+      this.logger.log(`Attempt for rebuy of current plan`)
       throw new ForbiddenException('You cannot rebuy a plan') // Actually I would love to allow it if it means more money
+    }
     if (paymentPlans[plan] < paymentPlans[user.paymentPlan]) {
+      this.logger.log(`Attempt for downgrad plan seen! Thats illegal!!!`)
```

Comment:
I mean, keep it

Reply:
we downgrade them, take their money, and then they upgrade, and we take it again